src/sif/autocost.cc

```diff
@@ -275,6 +275,9 @@ class AutoCost : public DynamicCost {
    * estimate is less than the least possible time along roads.
    */
   virtual float AStarCostFactor() const override {
+    if (fixed_speed_ != 0) {
+      return speedfactor_[fixed_speed_];
+    }
```

Comment:
rather than adding this complexity perhaps we should solve the problem of when people are  setting both of these and choose a winner. and whichever the winner is we should assign to `topspeed_` and let this code unchanged.

intuitively, when someone specifies `fixed_speed` then `top_speed` should get assigned the same value. does that make sense?

Reply:
I disagree. Fixed speed means only one speed. We want that disabled unless someone specified it.

Reply:
not sure what you disagree with haha I was just saying I agree with you, albeit not in a straight forward way

Reply:
sorry i disagree that we should assign top_speed to fixed_speed. if we do that then when someone says top_speed = 45, then we'll have fixed_speed = 45 and then every edge will be 45. maybe i am misunderstanding what you are saying? my assumption was that if fixed_speed is set, ie not 0, then we will use fixed_speed for every edge. so setting it to top_speed when top_speed is passed in will essentially break the whole top_speed feature. what am i missing here? :smile:

Reply:
ah maybe this is a language thing, let me write it in code, when i read what you said:

> assign top_speed to fixed_speed

my brain saw this code:

```c++
fixed_speed = top_speed;
```

Reply:
ah well yeah, I meant it the other way around :smile: `top_speed = fixed_speed` when both are passed

Reply:
@jbartrug you can revert this change and rather do `top_speed = fixed_speed` if `fixed_speed != kDefaultFixedSpeed` in `dynamiccost.h::get_base_costs()`